lib/src/Base/Solver/openturns/LsqSolver.hxx

```diff
@@ -0,0 +1,65 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Implementation class of an unbounded solver for systems of non-linear equations based on least square optimization
+ *
+ *  Copyright 2005-2024 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_LSQSOLVER_HXX
+#define OPENTURNS_LSQSOLVER_HXX
+
+#include "openturns/OTprivate.hxx"
+#include "openturns/SolverImplementation.hxx"
+
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class LsqSolver
+ *
+ * This class is an interface for the 1D nonlinear solverImplementations
+ */
+class OT_API LsqSolver :
```

Comment:
LsqSolver is probably ambiguous as we have already lots of LeastSquares* classes
maybe LeastSquaresEquationSolver ? LeastSquaresRootSolver ?
maybe @regislebrun has a better name for this

Reply:
I think we can go for LeastSquaresEquationsSolver, meaning that it's suitable also for a system of equations.
Question: to rename a class in the library, we use the same "customize.sh" that is used for module renaming?

Reply:
There is a `rename.sh` in `utils`

Usage :  `./utils/rename.sh OldClass NewClass `